### PR TITLE
netty-bom to provide resolved tcnative version

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
Netty's bom includes netty-tcnative dependencies with a variable for the
version [1]. However that variable isn't defined/resolved and therefore
leads to undefined dependencies.

[1] https://search.maven.org/artifact/io.netty/netty-bom/4.1.68.Final/pom

Modifications:
- Netty's bom file should inherit from netty-parent pom so variables can
  be resolved at inclusion time.

Result:
netty-bom allows for netty-tcnative version to be resolved in 3rd party
projects.

### CODE REVIEW

**Comment by Scottmitch** [comment]
@bondolo - ptal

File: bom/pom.xml [hunk 1]
```diff
@@ -69,206 +65,206 @@
...
         <artifactId>netty-transport-native-unix-common</artifactId>
-        <version>4.1.69.Final-SNAPSHOT</version>
+        <version>${project.version}</version>
         <classifier>linux-aarch_64</classifier>
       </dependency>
       <dependency>
         <groupId>io.netty</groupId>
         <artifactId>netty-transport-native-unix-common</artifactId>
-        <version>4.1.69.Final-SNAPSHOT</version>
+        <version>${project.version}</version>
```
Comment (normanmaurer) [inline bom/pom.xml:221]: These are adjusted by the maven release plugin and so should not be changed

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: bom/pom.xml (hunk 1 above)
Comment (Scottmitch) [reply bom/pom.xml:221]: can you clarify how this file is treated differently than other sub-projects that have `netty-parent` as their parent pom?

Comment (normanmaurer) [reply bom/pom.xml:221]: It shouldn't ... don't know why it hadn't used the netty-parent

Comment (Scottmitch) [reply bom/pom.xml:221]: If there is no difference seems preferable to keep it consistent with how other sub-projects use `${project.version}` (and other variables) as is done in this PR, no?

**Comment by Scottmitch** [comment]
build failure attributed to https://github.com/netty/netty/issues/11673

File: bom/pom.xml (hunk 1 above)
Comment (normanmaurer) [reply bom/pom.xml:221]: Yeah I guess ...

**Review by normanmaurer — APPROVED** [review]

**Comment by Scottmitch** [comment]
main (62a23f9e29)

**Comment by Scottmitch** [comment]
cherry-pick to main messed up indentation and check style is failing. fixed in https://github.com/netty/netty/pull/11676

File: bom/pom.xml [hunk 2]
```diff
@@ -17,17 +17,13 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>org.sonatype.oss</groupId>
-    <artifactId>oss-parent</artifactId>
-    <version>7</version>
-    <relativePath />
+    <groupId>io.netty</groupId>
+    <artifactId>netty-parent</artifactId>
```
Comment (bondolo) [inline bom/pom.xml:21]: The Google BOM best practices guide (https://jlbp.dev/JLBP-15) suggests not including the parent project if it has a `dependency` section in addition to the `dependencyManagement` section as those dependencies will be brought in to projects which import the BOM. I checked the "effective pom" for Netty parent, `mvn help:effective-pom` and it does include some direct dependencies which would be unexpectedly added to projects which import the BOM.

```
...
        <dependency>
          <groupId>io.projectreactor.tools</groupId>
          <artifactId>blockhound</artifactId>
          <version>1.0.3.RELEASE</version>
        </dependency>
      </dependencies>
    </dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.junit.jupiter</groupId>
        <artifactId>junit-jupiter-api</artifactId>
        <version>5.7.0</version>
        <scope>test</scope>
      </dependency>
...
```

Is there a way to switch back to the sonatype `oss-parent` parent pom and still have the versions resolve?

Comment (normanmaurer) [reply bom/pom.xml:21]: See https://github.com/netty/netty/pull/11682

**Pull request merged**